CMakeLists.txt

```diff
@@ -336,7 +336,7 @@ add_subdirectory(misc)
 # property which is set implicitly for custom command outputs
 include(misc/generated/CMakeLists.txt)
 
-if(EXISTS soc/${ARCH}/CMakeLists.txt)
+if(EXISTS ${ZEPHYR_BASE}/soc/${ARCH}/CMakeLists.txt)
```

Comment:
I'm actually not sure this is correct.
This worked w/o support of out of tree SoC definition.
With support of out of tree SoC, I think ${SOC_DIR} would be better than ${ZEPHYR_BASE}, but I I'm confused with the syntax: `add_subdirectory(dir1 dir2)` does it add dir1 and dir2 as subdirs?

In that case shouldn't we test CMakeLists.txt availability in dir1 and dir2 and then split the whole section into in tree and out of tree tests?

Reply:
should actually be
```
if(EXISTS ${SOC_DIR}/${ARCH}/CMakeLists.txt)
```

Reply:
ok, makes sense. Updated. Txs